Issue: https://github.com/tree-sitter/tree-sitter/issues/3623
An ERROR leaf can have zero recovery cost but still represents a syntax error.

--- a/src/node.c
+++ b/src/node.c
@@ -518,7 +518,7 @@
 }
 
 bool ts_node_has_error(TSNode self) {
-  return ts_subtree_error_cost(ts_node__subtree(self)) > 0;
+  return ts_node_is_error(self) || ts_subtree_error_cost(ts_node__subtree(self)) > 0;
 }
 
 bool ts_node_is_error(TSNode self) {
